src/content/app/species/SpeciesPage.tsx

```diff
@@ -18,24 +18,75 @@ import React from 'react';
 import { Helmet } from 'react-helmet-async';
 import loadable from '@loadable/component';
 
+import { useAppSelector } from 'src/store';
+
+import { getDisplayName } from 'src/shared/components/selected-species/selectedSpeciesHelpers';
+
+import { getGenomeByUrlId } from 'src/shared/state/genome/genomeSelectors';
+
 import useHasMounted from 'src/shared/hooks/useHasMounted';
 
+import { fetchGenomeInfo } from 'src/shared/state/genome/genomeApiSlice';
+import { getPathParameters, useUrlParams } from 'src/shared/hooks/useUrlParams';
+
+import type { ServerFetch } from 'src/routes/routesConfig';
+import type { AppDispatch } from 'src/store';
+
 const LoadableSpeciesPageContent = loadable(
   () => import('./SpeciesPageContent')
 );
 
 const SpeciesPage = () => {
   const hasMounted = useHasMounted();
+  const params = useUrlParams<'genomeId'>(['/species/:genomeId']);
+  const { genomeId: genomeIdInUrl } = params;
+
+  const species = useAppSelector((state) =>
+    getGenomeByUrlId(state, genomeIdInUrl as string)
+  );
+
+  const title = species
+    ? `${getDisplayName(species)} — Ensembl`
+    : 'Species page — Ensembl';
 
   return (
     <>
       <Helmet>
-        <title>Species page — Ensembl</title>
+        <title>{title}</title>
         <meta name="description" content="Species home page" />
       </Helmet>
       {hasMounted && <LoadableSpeciesPageContent />}
     </>
   );
 };
 
+export const serverFetch: ServerFetch = async (params) => {
+  const { path, store } = params;
+  const dispatch: AppDispatch = store.dispatch;
+  const { genomeId: genomeIdFromUrl } = getPathParameters<'genomeId'>(
+    '/species/:genomeId/',
+    path
+  );
+
+  if (!genomeIdFromUrl) {
+    return; // this won't happen; but makes typescript happy
+  }
```

Comment:
Can't `genomeIdFromUrl` be asserted as `string` at line 76 instead of this check?

Reply:
Good point.

✅